From 598dabc80038a9c4ed35c552d01a468efe025f28 Mon Sep 17 00:00:00 2001
From: Xu Mao <B34688@freescale.com>
Date: Mon, 21 Jul 2014 16:16:09 +0800
Subject: [PATCH 58/95] ENGR00323218 Movie Studio: Can't export files

Buffer header pointer from vpu component can only be
used as raw value, if use it to access structure member, segment fault occurs.
It might be IPC issue or limitation.

FRAMEDROP flag is only used in soft AVC encoder.  so it's safe to comment
out these lines when using vpu encoder.

Signed-off-by: Xu Mao <B34688@freescale.com>
---
 media/libstagefright/OMXCodec.cpp | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/media/libstagefright/OMXCodec.cpp b/media/libstagefright/OMXCodec.cpp
index 7b1603e..d97238a 100644
--- a/media/libstagefright/OMXCodec.cpp
+++ b/media/libstagefright/OMXCodec.cpp
@@ -2045,6 +2045,15 @@ void OMXCodec::on_message(const omx_message &msg) {
         {
             IOMX::buffer_id buffer = msg.u.extended_buffer_data.buffer;
 
+            /*  Buffer header pointer from vpu component can only be used as raw
+                value, if use it to access structure member, segment fault occurs.
+                It might be IPC issue or limitation.
+                
+                FRAMEDROP flag is only used in soft AVC encoder. so it's safe to
+                comment out these lines when using vpu encoder.
+            */
+            
+            /*
 			//If the frame is dropped by avcEncoder, need drop stamp in mDecodingTimeList
             OMX_BUFFERHEADERTYPE *head = (OMX_BUFFERHEADERTYPE *)buffer;
              if( mIsEncoder && mIsVideo && (!strcasecmp(MEDIA_MIMETYPE_VIDEO_AVC, mMIME)) &&
@@ -2052,6 +2061,7 @@ void OMXCodec::on_message(const omx_message &msg) {
                 CODEC_LOGV("drop time stamp %lld us", head->nTimeStamp);
                 getDecodingTimeUs();
              }
+             */
 
             CODEC_LOGV("EMPTY_BUFFER_DONE(buffer: %u)", buffer);
 
-- 
1.8.0

